### PR TITLE
Add `QueryProblemProgressEncoder`

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Add `QueryProblemProgressEncoder` to add the query problems to the history log for using it in the migration progress

### Linked issues

- [ ] Depends on #3266
- [ ] Depends on #3250

Progresses #3045
Breaks up #3112

### Functionality

- [x] modified existing workflow: `table-migration-experimental`

### Tests

- [x] modified unit tests
- [x] modified integration tests

### CODE REVIEW

File: tests/unit/progress/test_queries.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from databricks.labs.ucx.framework.owners import Ownership
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+from databricks.labs.ucx.progress.queries import QueryProblemProgressEncoder
+from databricks.labs.ucx.source_code.queries import QueryProblem
+
+
+@pytest.mark.parametrize(
+    "query_problem, failure",
+    [
+        (
```
Comment (nfx) [inline tests/unit/progress/test_queries.py:14]: inline to a variable within the function - it makes no sense to have a single-case paramatrised test.

**Review by nfx — APPROVED** [review]
lgtm

**Comment by JCZuurmond** [comment]
Do not merge this PR as `QueryProblem` progress encoder will become part of dashboard

**Comment by JCZuurmond** [comment]
Not including this anymore

**Pull request closed without merging**